lms/djangoapps/grades/tests/test_transformer.py

```diff
@@ -303,6 +362,32 @@ def test_course_version_collected_in_split(self):
         block_structure = get_course_blocks(self.student, blocks[u'course'].location, self.transformers)
         self.assertIsNotNone(block_structure.get_xblock_field(blocks[u'course'].location, u'course_version'))
 
+    @ddt.data(
+        (u'problem', u'capa.xml'),
+        (u'openassessment', u'openassessment.xml'),
+        (u'coderesponse', u'coderesponse.xml'),
+        (u'lti', u'lti.xml'),
+        (u'library_content', u'library_content.xml'),
+    )
+    @ddt.unpack
+    def test_different_problem_types(self, block_type, filename):
+        """
+        Test that transformation works for various block types
+        """
+        if block_type == u'library_content':
+            # Library content does not have a weight
+            metadata = {
+                u'graded': True,
+                u'due': datetime.datetime(2099, 3, 15, 12, 30, 0, tzinfo=pytz.utc),
+            }
+        else:
+            metadata = None  # Use the default
+        blocks = self.build_course_with_block_from_file(block_type, filename, metadata)
+        block_structure = get_course_blocks(self.student, blocks[u'course'].location, self.transformers)
+        sequence = block_structure[u'sequence']
+        subsection_factory = SubsectionGradeFactory(self.student, block_structure, blocks[u'course'])
+        subsection_factory.update(sequence)
```

Comment:
Generally you just run the code, and if it throws an exception the test fails.  You could do:
```python
def test_doesnt_raise_foo(self):
    try:
        dothing()
    except FooError as e:
        self.fail(e)
```
But I'm not sure if it's worth it.